testpmd/hooks/roles/mirror-rh-nfv/tasks/main.yml

```diff
@@ -0,0 +1,49 @@
+---
+- name: "Mirror index image"
+  shell:
+    cmd: >
+      skopeo copy 
+      --all
+      --authfile {{ mrn_pull_secret }}
+      --dest-tls-verify=false
+      --remove-signatures
+      docker://{{ mrn_index_image }}
+      docker://{{ mrn_mirror_index_image }}
+  register: copy
+  retries: 5
+  delay: 5
+  until:
+    - copy is not failed
+
+- name: "Create temp manifest dir"
+  tempfile:
+    state: directory
+  register: mrn_manifest_dir
+
+- name: "Mirror images from index"
+  shell:
+    cmd: >
+      oc adm catalog mirror
+      -a {{ mrn_pull_secret }}
+      --to-manifests={{ mnr_manifest_dir }}
+      {{ mrn_mirror_index_image }}
+      {{ mrn_local_registry }}
+
+- name: "Apply ImageContentSourcePolicy"
+  k8s:
+    src: "{{ mrn_manifest_dir }}/imageContentSourcePolicy.yaml"
```

Comment:
Note that adding a source policy will reboot all the nodes. It results in increased CI duration. In order to avoid reboot for this (to reduce the CI durtation), the top-level mirror entry is added as part of the installation itself.

Reply:
Just learned that the machine config operator will only reboot nodes on OCP-4.6 and below (https://github.com/openshift/machine-config-operator/pull/2259), that is 4.5 and 4.6.

Are you suggesting to add `quay.io/rh-nfv-int` to the installation to avoid one extra reboot on those OCP versions that need it? If so, I agree that it will speed-up the time, the only caveat I see is that during installation time we don't know yet if example-cnf will be executed next or not :thinking:.  Also, any other image not in that registry will also be added through the ImageContentSourcePolicy, for example dpdk.

Reply:
> Just learned that the machine config operator will only reboot nodes on OCP-4.6 and below (openshift/machine-config-operator#2259), that is 4.5 and 4.6.

Good to know about this improvement. I still believe machine config will apply the configs on all nodes (without reboot) and restart crio. Do you know whether nodes will go to NOT READY and then READY (faster)? How much time does it take (in 4.7)?

> Are you suggesting to add quay.io/rh-nfv-int to the installation to avoid one extra reboot on those OCP versions that need it? If so, I agree that it will speed-up the time, the only caveat I see is that during installation time we don't know yet if example-cnf will be executed next or not thinking. Also, any other image not in that registry will also be added through the ImageContentSourcePolicy, for example dpdk.

Having a mirror entry at the install time for unused images is not going to affect the deployment. All example cnf images are within the `quay.io/rh-nfv-int` repo, except testpmd image, for which there are already entries as it is from the redhat registry.

Reply:
>  I still believe machine config will apply the configs on all nodes (without reboot) and restart crio.

Correct, it still needs to restart the service *if* the  entries are not yet available in the `registries.conf`

> Do you know whether nodes will go to NOT READY and then READY (faster)? How much time does it take (in 4.7)?

Haven't measured in 4.7 but the last successful job https://www.distributed-ci.io/jobs/40c6303c-c6b7-4bdc-a78f-137b29380333/jobStates#95e71f77-2fb1-4b43-b185-f78565f2c225:file286 shows only  a few seconds when the nodes are checked.

> Having a mirror entry at the install time for unused images is not going to affect the deployment.

Do you know where I can add that entry so we can save an extra reboot/restart I think is in the file under the setting: `disconnected_registry_mirrors_file` but any pointer is appreciated.

Reply:
I'm including the org namespace (rh-nfv-int) in the installer trust bundles https://github.com/dci-labs/service-configs/pull/12 this should speed up when applying the ImageContentSourcePolicies in here

Reply:
> Haven't measured in 4.7 but the last successful job https://www.distributed-ci.io/jobs/40c6303c-c6b7-4bdc-a78f-137b29380333/jobStates#95e71f77-2fb1-4b43-b185-f78565f2c225:file286 shows only a few seconds when the nodes are checked.

Thanks, it's good to know.